frontend/test/metabase/query_builder/components/DataSelector.unit.spec.js

```diff
@@ -0,0 +1,348 @@
+import React from "react";
+import { render, cleanup, fireEvent } from "@testing-library/react";
+
+import _ from "underscore";
+
+import { delay } from "metabase/lib/promise";
+
+import {
+  SAMPLE_DATASET,
+  ANOTHER_DATABASE,
+  MULTI_SCHEMA_DATABASE,
+  metadata,
+  makeMetadata,
+  state as fixtureData,
+} from "__support__/sample_dataset_fixture";
+
+import { UnconnectedDataSelector as DataSelector } from "metabase/query_builder/components/DataSelector";
+
+describe("DataSelector", () => {
+  let originalTimeout;
+  beforeEach(() => {
+    originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000;
```

Comment:
Why so long?

Reply:
This one was weird. I still don't know the cause. Without the increased timeout, this suite passes alone but consistently fails when running all of them. Googling indicated that the timeout error can be due to conflict between Promise libraries. I tried running different subsets of suites and there is some "tipping point", but it's not consistently any one file that clashes.

I have no idea what would take more than five seconds, but bumping it seemed to make it consistently pass. 🤷‍♂